rules/TypeDeclaration/Rector/ClassMethod/ReturnNeverTypeRector.php

```diff
@@ -118,18 +114,18 @@ private function shouldSkip(ClassMethod | Function_ | Closure $node, Scope $scop
             return true;
         }
 
+        if ($node->returnType instanceof Node && $this->isName($node->returnType, 'void')) {
+            return false;
+        }
+
         if ($node instanceof ClassMethod && $this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod(
             $node,
             $scope
         )) {
             return true;
         }
 
-        if (! $node->returnType instanceof Node) {
-            return false;
-        }
-
-        return $this->isName($node->returnType, 'never');
```

Comment:
no need for this check, as we don't need to change the return type to `never` when it is already `never`